### PR TITLE
avoid iteration when possible in `each_proxy`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/each_proxy.js [hunk 1]
```diff
@@ -44,8 +44,8 @@ class EachProxy {
   arrayWillChange(content, idx, removedCnt, addedCnt) {   // eslint-disable-line no-unused-vars
     let keys = this._keys;
     let lim = removedCnt > 0 ? idx + removedCnt : -1;
-    for (let key in keys) {
-      if (lim > 0) {
+    if (lim > 0) {
+      for (let key in keys) {
```
Comment (bekzod) [inline packages/ember-metal/lib/each_proxy.js]: 🎈

File: packages/ember-metal/lib/each_proxy.js [hunk 2]
```diff
@@ -44,8 +44,8 @@ class EachProxy {
   arrayWillChange(content, idx, removedCnt, addedCnt) {   // eslint-disable-line no-unused-vars
     let keys = this._keys;
     let lim = removedCnt > 0 ? idx + removedCnt : -1;
-    for (let key in keys) {
-      if (lim > 0) {
```
Comment (bekzod) [inline packages/ember-metal/lib/each_proxy.js]: 🎈

**Review by krisselden — APPROVED** [review]

**Comment by bekzod** [comment]
needs restart

**Comment by krisselden** [comment]
looks like IE crashed
image

**Pull request merged**